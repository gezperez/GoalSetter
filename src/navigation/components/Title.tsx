import React, { FC } from 'react'
import { StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'

import Typography from '~/utils/Typographys'

type TitleProps = {
  title: string
  subtitle?: string
}

const Title: FC<TitleProps> = ({ title, subtitle }) => (
  <View style={styles.container}>
    <Text style={styles.title}>{title}</Text>
    {subtitle && <Text style={styles.subtitle}>{subtitle}</Text>}
  </View>
)

type Styles = {
  container: ViewStyle
  title: TextStyle
  subtitle: TextStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: Typography.headerTitle,
  subtitle: {
    ...Typography.headerSubtitle,
    textAlign: 'center',
    marginTop: 10,
  },
})

export default Title
