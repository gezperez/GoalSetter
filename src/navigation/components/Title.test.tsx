import React from 'react'
import { render } from '@testing-library/react-native'

import I18nJs from '~/__mocks__/react-native-i18n'
import Title from './Title'

describe('Title', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const { toJSON, getByText } = render(<Title title={I18nJs.t('signUpHeaderTitle')} />)

    const title = getByText(I18nJs.t('signUpHeaderTitle'))

    expect(title).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with non required props', () => {
    const { toJSON, getByText } = render(
      <Title title={I18nJs.t('signUpHeaderTitle')} subtitle={I18nJs.t('signUpHeaderSubtitle')} />
    )

    const title = getByText(I18nJs.t('signUpHeaderTitle'))
    const subtitle = getByText(I18nJs.t('signUpHeaderSubtitle'))

    expect(title).toBeDefined()
    expect(subtitle).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })
})
