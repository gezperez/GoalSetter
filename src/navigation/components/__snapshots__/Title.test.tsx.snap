// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render with non required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 20,
        "lineHeight": 23,
      }
    }
  >
    [missing "en.signUpHeaderTitle" translation]
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Light",
        "fontSize": 14,
        "lineHeight": 16,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    [missing "en.signUpHeaderSubtitle" translation]
  </Text>
</View>
`;

exports[`Title should render with required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 20,
        "lineHeight": 23,
      }
    }
  >
    [missing "en.signUpHeaderTitle" translation]
  </Text>
</View>
`;
